### DISCUSSION TITLE
Why doesn't /delete-node/ work inside of a referenced node?

### DISCUSSION THREAD

**Discussion by cdwilson** [zephyrproject-rtos/zephyr#67228] [Q&A] [answered]
I'm working on a board with a device tree that contains the following definition:

```
&spi3 {
	compatible = "nordic,nrf-spim";
	status = "okay";
	cs-gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
	pinctrl-0 = <&spi3_default>;
	pinctrl-1 = <&spi3_sleep>;
	pinctrl-names = "default", "sleep";
	w25q32jv: w25q32jv@0 {
		compatible = "jedec,spi-nor";
		reg = <0>;
		spi-max-frequency = <40000000>;
		wp-gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
		hold-gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
		size = <0x2000000>;
		has-dpd;
		t-enter-dpd = <3000>;
		t-exit-dpd = <30000>;
		jedec-id = [ ef 40 16 ];
	};
};
```

As I mentioned in https://github.com/zephyrproject-rtos/zephyr/discussions/35214#discussioncomment-5795369 I discovered it is possible to delete the `w25q32jv` node from within an overlay by placing the `/delete-node/` **_outside_** of the `&spi3` referenced node in the overlay:

```
/delete-node/ &w25q32jv; <-- this works

&spi3 {
	cs-gpios = <&feather_header 15 GPIO_ACTIVE_LOW>;
```

However, placing the `/delete-node/` **_inside_** of the `&spi3` referenced node does not work:

```
&spi3 {
    cs-gpios = <&feather_header 15 GPIO_ACTIVE_LOW>;

    /delete-node/ w25q32jv; <-- this does not work
```

Can someone explain why the `/delete-node/` only works if placed outside of the reference block in an overlay? It doesn't really make sense to me why one works but the other does not.

**Comment by nordicjm** [comment, marked as answer]
Because you are not using the inner node, you would need to do `/delete-node/ w25q32jv@0;`

**Reply by cdwilson** [reply]
Thanks @nordicjm!

For others running across this in the future, here are some additional examples:

```
/ {
	/* this works */
	soc {
		peripheral@40000000 {
			spi@b000 {
				/delete-node/ w25q32jv@0;
			};
		};
	};

	/* this does NOT work (compiles, but node is not deleted) */
	soc {
		peripheral@40000000 {
			spi@b000 {
				/delete-node/ w25q32jv;
			};
		};
	};
};

&spi3 {
	/* this works */
	/delete-node/ w25q32jv@0;

	/* this does NOT work (compiles, but node is not deleted) */
	/delete-node/ w25q32jv;
};

/* this works */
/delete-node/ &w25q32jv;

/* this works */
/delete-node/ &{/soc/peripheral@40000000/spi@b000/w25q32jv@0};

```

**Comment by Lets-a-go** [comment]
This is a fairly old discussion, but still very relevant. A word of warning is in order, however:

Using /delete-node/ in an overlay can cause unanticipated devicetree configurations. 

For example, if you delete a high level node such as &pinctrl in an overlay file, attempting to create your own pin configs in a higher-precedence overlay file, any lower-presence overlay changes to &pinctrl will vanish, even though the lower precedence file exists. Of course, that sort of override may be your exact intention. Take care using it. Try to stay away from blanket deletions of higher level nodes than needed to avoid unanticipated side-effects. Try to target the deepest node that has just what you need to change. This is a powerful tool.
